Source/WebCore/page/EventHandler.cpp

```diff
@@ -1669,10 +1669,10 @@ std::optional<Cursor> EventHandler::selectCursor(const HitTestResult& result, bo
             && !m_mouseDownMayStartDrag
 #endif
             && frame->selection().isCaretOrRange()
-            && !m_capturingMouseEventsElement)
+            && !m_capturingMouseEventsElement &&  !(renderer->style().usedUserSelect()  == UserSelect::None))
```

Comment:
Nit - extra spaces. I'd also use `!=` rather than `!(… == …)`

```suggestion
            && !m_capturingMouseEventsElement && renderer->style().usedUserSelect()  != UserSelect::None)
```